From dee7b5d37ee46bd665ff2cebf26b83d11ef266c6 Mon Sep 17 00:00:00 2001
From: nushanth <nushanth@qti.qualcomm.com>
Date: Mon, 13 Jul 2026 19:47:32 +0530
Subject: [PATCH] Agent: Fix operating class entries in Operating Channel
 Report

Avoid adding both primary and center channels for operating classes.
Make the primary-channel logic a fallback to the center-channel path,
ensuring each operating class is reported only once with the correct
channel value.

Signed-off-by: nushanth <nushanth@qti.qualcomm.com>
---
 agent/src/beerocks/slave/tlvf_utils.cpp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/agent/src/beerocks/slave/tlvf_utils.cpp b/agent/src/beerocks/slave/tlvf_utils.cpp
index cdb3032..9ef2d54 100644
--- a/agent/src/beerocks/slave/tlvf_utils.cpp
+++ b/agent/src/beerocks/slave/tlvf_utils.cpp
@@ -330,11 +330,6 @@ bool tlvf_utils::create_operating_channel_report(ieee1905_1::CmduMessageTx &cmdu
         for (auto it = operating_classes.begin();
              it != operating_classes.end() && *it <= operating_class; ++it) {

-            if (son::wireless_utils::is_channel_in_operating_class(
-                    *it, radio->wifi_channel.get_channel())) {
-                operating_list.emplace_back(std::make_pair(*it, radio->wifi_channel.get_channel()));
-            }
-
             if (son::wireless_utils::is_operating_class_using_central_channel(*it)) {
                 std::map<uint8_t, std::map<beerocks::eWiFiBandwidth, son::wireless_utils::sChannel>>
                     channels_table;
@@ -407,6 +402,9 @@ bool tlvf_utils::create_operating_channel_report(ieee1905_1::CmduMessageTx &cmdu

                 operating_list.emplace_back(
                     std::make_pair(*it, center_channel_it->second.center_channel));
+            } else if (son::wireless_utils::is_channel_in_operating_class(
+                    *it, radio->wifi_channel.get_channel())) {
+                operating_list.emplace_back(std::make_pair(*it, radio->wifi_channel.get_channel()));
             }
         }

--
2.34.1

